src/late_multicellular_stage/Scalis.cs

```diff
@@ -0,0 +1,309 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   Mathematic function based on convolution surfaces. Uses bones, each defined by two metaballs, to calculate value
+///   of an arbitrary point in space.
+/// </summary>
+public class Scalis : IMeshGeneratingFunction
+{
+    public float CutoffPointMultiplier = 1.5f;
+
+    public float InnerCutoffPointMultiplier = 2.0f;
+
+    public bool Cutoff = true;
+
+    private static int[] coefficients = { 1, 2, 1 };
+
+    /// <summary>
+    ///   Every metaball represents a point in creature's skeleton.
+    ///   Each bone is formed between a point and its parent.
+    /// </summary>
+    private readonly MulticellularMetaball[] points;
+
+    private float surfaceValue = 1.0f;
+
+    public Scalis(IReadOnlyCollection<MulticellularMetaball> metaballs)
+    {
+        points = new MulticellularMetaball[metaballs.Count];
+
+        int i = 0;
+
+        foreach (var point in metaballs)
+        {
+            points[i] = point;
+            ++i;
+        }
+    }
+
+    public float SurfaceValue
+    {
+        get => surfaceValue;
+        set => surfaceValue = value;
+    }
+
+    public float GetValue(Vector3 pos)
+    {
+        const float sigma = 1.0f;
+
+        float value = 0.0f;
+
+        pos /= sigma;
```

Comment:
I'll assume that changing this to `pos *= 1 / sigma` is the equivalent (not counting floating point imprecision). As division is a much more expensive operation, I'll try swapping all the divisions that are easy to convert to multiplications here.